packages/blade/src/components/Collapsible/_decisions/decisions.md

```diff
@@ -0,0 +1,186 @@
+
+![Collapsible breakdown](./collapsible-breakdown.png)
+
+```jsx
+import { Collapsible, CollapsibleButton, CollapsibleLink, CollapsibleBody } from '@razorpay/blade';
+
+<Collapsible>
+  <CollapsibleButton>Button</CollapsibleButton>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+
+<Collapsible>
+  <CollapsibleLink>Button</CollapsibleLink>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+```
+
+> **Note**
+>
+> - `Collapsible` will compose (`CollapsibleButton` or `CollapsibleLink`), `CollapsibleBody`. This ensures all interactions, a11y and animations work seamlessly at user end.
+> - `CollapsibleButton` and `CollapsibleLink` will support a subset of relevant props from `Button` and `Link` respectively.
+
+### Collapsible
+
+| Prop              | Type            | Default     | Description                                                        | Required |
+| ----------------- | --------------- | ----------- | ------------------------------------------------------------------ | -------- |
+| children          | `JSX`           | `undefined` | Composes `CollapsibleButton`, `CollapsibleLink`, `CollapsibleBody` | ✅       |
+| direction         | `bottom`, `top` | `bottom`    | Direction in which the content expands                             |          |
+| defaultIsExpanded | `boolean`       | `true`      | Expands the collapsible content by default (uncontrolled)          |          |
+| isExpanded        | `boolean`       | `undefined` | Expands the collapsible content (controlled)                       |          |
+| onChange          | `function`      | `undefined` | Callback for change in collapsible's expanded state                |          |
+
+> **Note**
+>
+> - Also includes layout based styling props
+> - For using collapsible in controlled state, use `isExpanded` coupled with `onChange`
+> - `onChange` callback signature `({ isExpanded }) => {}`, `isExpanded` represents the collapsible's current state
+
+### CollapsibleButton
+
+Built-in trigger for `Collapsible`, provides a subset of `Button` [properties](../../Button/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `variant`
+- `size`
+- `icon`
+- `iconPosition`
+- `isDisabled`
+- `testID`
+- `accessibilityLabel`
+
+Doesn't support ❌
+
+- `isFullWidth` - can visually break UI
+- `onClick`, `isLoading` - using `onClick` will break built-in interactions (unless we provide additional hooks), `onChange` should already provide a way for users to listen-in on `Collapsible` state. This might have usage for async with `isLoading` in the future but for now not exposing the prop
+- `type` - will be `button` always
+
+### CollapsibleLink
+
+Built-in trigger for `Collapsible`, provides a subset of `Link` [properties](../../Link/_decisions/decisions.md).
+
+Supports ✅
+
+- `children`
+- `icon`
```

Comment:
Consumers will have to handle changing of icon here, right? So no animation of the icon right?

Reply:
In that case should we restrict users from passing icons entirely? We'll handle the icon and its animation then for both Button & Link variants?

Reply:
Discussed with design, here's the outcome on dev API:

- We’ll let users pass all the props (including icons) (on Figma side there isn't a way to restrict this)
- We’ll not have right icon in link button trigger (because chevron takes that place)
- The chevron icon will animate (custom icons will not)
- No provision to disable rotating chevron icon on link button (note: button has no rotating chevron)